### ISSUE TITLE
Change zip code filter on filter modal to region filter

### ISSUE THREAD

**Issue by jnowell** [CityOfDetroit/bloom#814]
After user testing, most users didn't find the Zip Code filter particularly useful. After some internal discussion, we think what makes the most sense is to have a filter that allows you to filter by a particular region. The regions will be defined by [this district map](https://commons.wikimedia.org/wiki/File:Detroit_districts_map.png), with the neighborhood -> region mapping to be defined by Tara. Once the mapping is completed, I'll update this bug. The filter should behave the same as the Accessibility Filter, with multi-select checkboxes (so that the user can include multiple regions in their search). You can use the same design we use for Accessible Features in [Figma](https://www.figma.com/file/wAhoXVmHRIJ2TyBKFVq2JY/Desktop-and-Mobile-Experiences?node-id=3540%3A13497) (02.2). 
![zip_code_listings](https://user-images.githubusercontent.com/394908/145111004-3e874bc3-fe1c-48a3-8b0a-354c7a2919c1.png)

**Comment by GabeBenjamin** [comment]
This is a more involved change (we need to first add the ability to filter based on neighborhood, then make a mapping of areas to neighborhoods) and we will probably not be able to get this in by the meeting with the mayor.
